scripts/controlnet.py

```diff
@@ -739,16 +739,32 @@ def process(self, p, *args):
                 detected_map = np.ndarray((round(input_image.shape[0]/4),input_image.shape[1]),dtype="float32",buffer=detected_map_bytes)
                 detected_map = torch.Tensor(detected_map).to(devices.get_device_for("controlnet"))
                 is_image = False
-                            
+
+            if issubclass(type(p), StableDiffusionProcessingTxt2Img) and p.enable_hr:
```

Comment:
Consider using `isinstance`, i.e.:
```suggestion
            if isinstance(p, StableDiffusionProcessingTxt2Img) and p.enable_hr:
```

Reply:
Sure!